pkg/aspect/config/attrgen.proto

```diff
@@ -0,0 +1,38 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package pkg.aspect.config;
+
+import "gogoproto/gogo.proto";
+
+option go_package="config";
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// Configures an AttributeGenerator aspect.
+message AttributeGeneratorsParams {
+    // Map of input params name to attribute expressions. At runtime, each
+    // expression will be evaluated to determine the input value provided to the
+    // aspect.
+    map<string, string> input_expressions = 1;
+
+    // Map of generated value names to mixer attribute names. Individual aspect
+    // implementations will produce a map of output values (name -> value).
+    // These output values will be converted into attributes for consumption by
+    // the rest of the mixer via this map.
+    map<string, string> value_attribute_map = 2;
```

Comment:
This was how the original PR was setup. After discussions with @mandarjog, we decided it was better to have the descriptors be part of the global config and have their names referenced in the aspect config.

This choice makes it so that adapters don't have to know anything about how they are mapped into descriptors and so that operators can look at global config to find the full definition of attributes. I believe that having the descriptors be global matches the rest of the config story and makes validation easier too.